README.md

```diff
@@ -1,22 +1,57 @@
 
-The TAF is based on py.test (http://pytest.org/latest/). The basic concept is to create library with all necessary functionality, to define how the environment configuration will be supplied to test suites, and to create reporting processing functionality.
+## TAF overview
 
-TAF package consists of two folders: ./taf-repo and ./testcases-repo. taf-repo folder holds library modules (or plugins) for py.test, and testcases-repo contains test cases including test configuration and test scenarios. ./taf-repo/testlib contains of the following modules.
+**TAF** is an integrated cross-platform system that sets the guidelines and provides tools for automation testing of a specific product in Networking. The function libraries, test data sources, test object details and various reusable modules are integrated in one system. The platform provides an outline of test automation process and reduces the automation efforts. With this framework, automation or validation engineer may run testing suites on different environments and obtain the test results with detail reports.
 
-## Common functionality and helper functions:
-* loggers - integration of standard python logging library into Framework;
-* helpers - contains different helper functions which might be used from test suites;
-* common3 - contains base class to handle test environment.
+The [Python](https://www.python.org/) programming language was chosen to the Framework, mostly taking into account high development speed and great maintainability. It has an XML-RPC library. [Py.test](http://pytest.org/latest/) has several advantages over other available test libraries. We follow the principle of maximum usage of existing components and systems. The framework has some components which use threading which should be considered when looking for components to extend the system. The system is built in way that allows users to see all errors that occur during test execution and to trace the full path to the initial point where the defect appeared to simplify root cause analysis. There is no external storage required except integration with Test Case Management System(s) using a database via API to update test case execution results and any related information (automation status, automation owner, automated script location, etc.)
 
-## Groups of modules to handle environment-specific functionality:
-* environment - functions for manipulation with operating system environment variables etc.
+### Design considerations
 
-## Hardware-related libraries:
-* dev_* - modules which contain classes to handle particular environment type;
-* powerboard - startup/shutdown of the devices using SNMP via APC power boards;
+**System Requirements and Dependencies**
+
+The framework is designed for use in Linux systems which have Python programming tools and the necessary libraries installed. The framework has been tested on following systems:
+
+* Ubuntu 12.04 LTS running Python 3.4.0, py.test 2.9.2 and higher versions.
```

Comment:
I will remove mentions about Ubuntu 12.04 in README and html docs in next commit.